src/components/Capture/withCameraDetection.js

```diff
@@ -0,0 +1,25 @@
+import { h } from 'preact'
+import { PureComponent } from 'preact-compat'
+import { checkIfHasWebcam } from '../utils'
+
+export default WrappedComponent =>
+  class WithCameraDetection extends PureComponent {
```

Comment:
why `PureComponent`, isn't this not a pure component?

Reply:
I guess this definition comes from more recent versions of React, where pure components will conditionally render if shallow-comparing `nextProps` or `nextState` turns out to be different during `shouldComponentUpdate`

Since we have a `setInterval` here, even if we constantly call `this.setState({ hasCamera: true })` it will trigger a re-render in a regular component, and won't on a pure component.

I think this is confusing because we used `CameraPure` and `UploaderPure` to name stateless functional components, but those aren't really pure in the sense they have that check.